index.js

```diff
@@ -56,11 +56,14 @@ function setup() {
     version,
     args["package-manager"]
   );
-  const framework = packageJson.dependencies["@remix-run/react"]
-    ? "remix"
-    : packageJson.dependencies["react-router"]
-    ? "react-router"
-    : null;
+  const framework =
+    packageJson.dependencies["@remix-run/react"] ||
+    packageJson?.devDependencies["@remix-run/react"]
+      ? "remix"
+      : packageJson.dependencies["react-router"] ||
+        packageJson?.devDependencies["react-router"]
+      ? "react-router"
+      : null;
```

Comment:
```suggestion
  const framework =
    packageJson.dependencies["@remix-run/react"] ||
    packageJson.devDependencies["@remix-run/react"]
      ? "remix"
      : packageJson.dependencies["react-router"] ||
        packageJson.devDependencies["react-router"]
      ? "react-router"
      : null;
```

Reply:
We shouldn't need the defensiveness here

Reply:
The package.json might not have a devDependecies field, so that's why I used the optional chaining operator.

Reply:
I think you would want `packageJson.devDependencies?.["react-router"]` then? 

Let's DRY this up and just do this which will allow for either `dependencies` or `devDepoendencies` to be non-existent:

```suggestion
  const allDeps = { ...packageJson.dependencies, ...packageJson.devDependencies };
  const framework = 
    allDeps["@remix-run/react"] ? "remix" : 
    allDeps["react-router"] ? "react-router" : 
    null;
```

Reply:
Whoops, you are correct. Good call! LGTM.